kt/api-generator/src/main/kotlin/godot/codegen/services/impl/AwaitGenerationService.kt

```diff
@@ -0,0 +1,131 @@
+    override fun generate(maxArgumentCount: Int): FileSpec {
+        val awaitFile = FileSpec.builder(godotCoroutinePackage, "Await")
+
+        val allParameters = Array(maxArgumentCount) { index ->
+            TypeVariableName("P$index")
+        }.toList()
+
+        for (argCount in 0..maxArgumentCount) {
+            val parameters = allParameters.take(argCount)
+
+            val baseReceiver = ClassName(godotCorePackage, signal + argCount)
+            val receiver = if (argCount != 0) {
+                baseReceiver.parameterizedBy(parameters)
+            } else {
+                baseReceiver
+            }
+
+            val returnType: TypeName = when (argCount) {
+                0 -> UNIT
+                1 -> parameters[0]
+                else -> ClassName(godotCoroutinePackage, "SignalArguments$argCount").parameterizedBy(parameters)
+            }
+
+            // Create a tuple for the signal arguments
+            if (argCount >= 2) {
+                awaitFile.addType(
+                    TypeSpec.classBuilder("SignalArguments$argCount")
+                        .addModifiers(KModifier.DATA)
+                        .primaryConstructor(
+                            FunSpec
+                                .constructorBuilder()
+                                .apply {
+                                    parameters.forEachIndexed { index, parameter ->
+                                        addParameter("p$index", parameter)
+                                    }
+                                }
+                                .build()
+                        )
+                        .addProperties(
+                            parameters.mapIndexed { index, parameter ->
+                                PropertySpec
+                                    .builder("p$index", parameter)
+                                    .initializer("p$index")
+                                    .build()
+                            }
+                        )
+                        .addTypeVariables(parameters)
+                        .build()
+                )
+            }
+
+            awaitFile.addFunction(
+                FunSpec.builder("await")
+                    .addModifiers(KModifier.SUSPEND, KModifier.INLINE)
+                    .receiver(receiver)
+                    .apply {
+                        if (argCount != 0) {
+                            addTypeVariables(parameters.map { it.copy(reified = true) })
+                        }
+                    }
+                    .generateBody(argCount, returnType)
+                    .returns(returnType)
+                    .build()
+            )
+        }
+
+        return awaitFile
+            .addAnnotation(
+                AnnotationSpec.builder(ClassName("kotlin", "Suppress"))
+                    .addMember("\"PackageDirectoryMismatch\", \"unused\"")
+                    .build()
+            )
+            .indent("    ")
```

Comment:
Why is this needed ?

Reply:
Because Kotlin Poet was using indent of 2 spaces, and somehow was failing the formatting of the code, which led to it not compiling.